tests/MongoDB.Driver.Core.Tests/Core/Operations/OperationTestBase.cs

```diff
@@ -247,9 +248,10 @@ protected IReadBinding CreateReadBinding(ReadPreference readPreference)
             return new ReadPreferenceBinding(_cluster, readPreference, _session.Fork());
         }
 
-        protected IReadWriteBinding CreateReadWriteBinding()
+        protected IReadWriteBinding CreateReadWriteBinding(bool useExplicitSessionIfSupported = true)
         {
-            return new WritableServerBinding(_cluster, _session.Fork());
+            var session = CreateSession(useExplicitSessionIfSupported, _cluster);
+            return new WritableServerBinding(_cluster, session.Fork());
```

Comment:
Only call Fork if someone else owns the session. Here we just created the session and we own it so no need to call Fork.